src/popper/runner_host.py

```diff
@@ -193,7 +193,23 @@ def _create_container(self, cid, step):
         elif not self._config.skip_pull and not step.skip_pull:
             log.info(f"[{step.id}] docker pull {img}:{tag}")
             if not self._config.dry_run:
-                self._d.images.pull(repository=f"{img}:{tag}")
+                streamer = self._d.api.pull(
+                    repository=f"{img}:{tag}", decode=True,
+                )
+                
+                for chunk in streamer.splitlines():
+                    if self._config.quiet:
+                        continue
+                    chunk = chunk.strip()
+                    if chunk:
+                        import json
+                        chunk = json.loads(chunk)
+                        if 'id' in chunk:
+                            log.step_info(f"[{step.id}] " + chunk['id'] + ': ' + chunk['status'])
```

Comment:
to be consistent with the way logging is done in other places, this can ommit the `[{step.id}] ` prefix and log the output directly